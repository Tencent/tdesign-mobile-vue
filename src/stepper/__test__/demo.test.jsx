/**
 * 该文件为由脚本 `npm run test:demo` 自动生成，如需修改，执行脚本命令即可。请勿手写直接修改，否则会被覆盖
 */

import { mount } from '@vue/test-utils';
import baseVue from '@/stepper/demos/base.vue';
import minMaxVue from '@/stepper/demos/min-max.vue';
import mobileVue from '@/stepper/demos/mobile.vue';
import sizeVue from '@/stepper/demos/size.vue';
import statusVue from '@/stepper/demos/status.vue';
import themeVue from '@/stepper/demos/theme.vue';

const mapper = {
  baseVue,
  minMaxVue,
  mobileVue,
  sizeVue,
  statusVue,
  themeVue,
};

describe('Stepper', () => {
  Object.keys(mapper).forEach((demoName) => {
    it(`Stepper ${demoName} demo works fine`, () => {
      const wrapper = mount(mapper[demoName]);
      expect(wrapper.element).toMatchSnapshot();
    });
  });
});
