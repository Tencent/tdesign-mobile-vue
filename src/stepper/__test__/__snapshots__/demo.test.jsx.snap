// Vitest Snapshot v1

exports[`Stepper > Stepper baseVue demo works fine 1`] = `
<div
  class="stepper-demo"
>
  <div
    class="t-cell t-cell--middle t-cell--bordered"
  >
    <div
      class="t-cell__left-icon"
    >
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__title"
    >
      标题文字
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__note"
    >
      
      <div
        class="t-stepper"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          type="tel"
        />
        <div
          class="t-stepper__plus"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`Stepper > Stepper disableInputVue demo works fine 1`] = `
<div
  class="stepper-demo"
>
  <div
    class="t-cell t-cell--middle t-cell--bordered"
  >
    <div
      class="t-cell__left-icon"
    >
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__title"
    >
      禁止输入
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__note"
    >
      
      <div
        class="t-stepper"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          disabled=""
          readonly=""
          type="tel"
        />
        <div
          class="t-stepper__plus"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`Stepper > Stepper disabledVue demo works fine 1`] = `
<div
  class="stepper-demo"
>
  <div
    class="t-cell t-cell--middle t-cell--bordered"
  >
    <div
      class="t-cell__left-icon"
    >
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__title"
    >
      禁用
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__note"
    >
      
      <div
        class="t-stepper t-is-disabled"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          disabled=""
          type="tel"
        />
        <div
          class="t-stepper__plus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      
    </div>
    <!--v-if-->
  </div>
</div>
`;

exports[`Stepper > Stepper eventVue demo works fine 1`] = `
<div
  class="stepper-demo"
>
  <div
    class="t-cell t-cell--middle t-cell--bordered"
  >
    <div
      class="t-cell__left-icon"
    >
      <!--v-if-->
      <!--v-if-->
    </div>
    <div
      class="t-cell__title"
    >
      标题文字（单位）
      <!--v-if-->
      <!--v-if-->
    </div>
    <!--v-if-->
    <div
      class="t-cell__right-icon"
    >
      
      <div
        class="t-stepper"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          type="tel"
        />
        <div
          class="t-stepper__plus"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      
    </div>
  </div>
</div>
`;

exports[`Stepper > Stepper mobileVue demo works fine 1`] = `
<div
  class="tdesign-mobile-demo"
>
  <h1
    class="title"
  >
    Stepper 步进器
  </h1>
  <p
    class="summary"
  >
    用户通过调整“+”按钮、“-”按钮、数字输入框来调整具体需要的数值，可设置最大值和最小值
  </p>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      01 类型
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      基础步进器
    </p>
    
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          标题文字
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__note"
        >
          
          <div
            class="t-stepper"
          >
            <div
              class="t-stepper__minus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              type="tel"
            />
            <div
              class="t-stepper__plus"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
        <!--v-if-->
      </div>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      带单位步进器
    </p>
    
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          标题文字（单位）
          <!--v-if-->
          <!--v-if-->
        </div>
        <!--v-if-->
        <div
          class="t-cell__right-icon"
        >
          
          <div
            class="t-stepper"
          >
            <div
              class="t-stepper__minus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              type="tel"
            />
            <div
              class="t-stepper__plus"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <!--v-if-->
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      纯步进器
    </p>
    
    <div
      style="margin-left: 16px;"
    >
      <div
        class="t-stepper t-stepper__pure"
      >
        <div
          class="t-stepper__minus"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          type="tel"
        />
        <div
          class="t-stepper__plus"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    
  </div>
  <div
    class="tdesign-mobile-demo-block"
  >
    <h2
      class="tdesign-mobile-demo-block__title"
    >
      02 类型
    </h2>
    <p
      class="tdesign-mobile-demo-block__summary"
    >
      步进器状态
    </p>
    
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          禁用
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__note"
        >
          
          <div
            class="t-stepper t-is-disabled"
          >
            <div
              class="t-stepper__minus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              disabled=""
              type="tel"
            />
            <div
              class="t-stepper__plus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
        <!--v-if-->
      </div>
    </div>
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          禁止输入
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__note"
        >
          
          <div
            class="t-stepper"
          >
            <div
              class="t-stepper__minus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              disabled=""
              readonly=""
              type="tel"
            />
            <div
              class="t-stepper__plus"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
        <!--v-if-->
      </div>
    </div>
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          最大值
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__note"
        >
          
          <div
            class="t-stepper"
          >
            <div
              class="t-stepper__minus"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              type="tel"
            />
            <div
              class="t-stepper__plus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
        <!--v-if-->
      </div>
    </div>
    <div
      class="stepper-demo"
    >
      <div
        class="t-cell t-cell--middle t-cell--bordered"
      >
        <div
          class="t-cell__left-icon"
        >
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__title"
        >
          最小值
          <!--v-if-->
          <!--v-if-->
        </div>
        <div
          class="t-cell__note"
        >
          
          <div
            class="t-stepper"
          >
            <div
              class="t-stepper__minus t-is-disabled"
            >
              <svg
                class="t-icon t-icon-remove t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M3.5 7.35h9v1.3h-9v-1.3z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
            <input
              class="t-stepper__input"
              type="tel"
            />
            <div
              class="t-stepper__plus"
            >
              <svg
                class="t-icon t-icon-add t-stepper__icon"
                fill="none"
                height="1em"
                viewBox="0 0 16 16"
                width="1em"
              >
                <path
                  d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
                  fill="currentColor"
                  fill-opacity="0.9"
                />
              </svg>
            </div>
          </div>
          
        </div>
        <!--v-if-->
      </div>
    </div>
    <div
      class="stepper-inline-demo"
    >
      <div
        class="t-stepper t-is-disabled t-stepper__pure"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          disabled=""
          type="tel"
        />
        <div
          class="t-stepper__plus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-stepper t-stepper__pure"
      >
        <div
          class="t-stepper__minus"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          type="tel"
        />
        <div
          class="t-stepper__plus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
      <div
        class="t-stepper t-stepper__pure"
      >
        <div
          class="t-stepper__minus t-is-disabled"
        >
          <svg
            class="t-icon t-icon-remove t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M3.5 7.35h9v1.3h-9v-1.3z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
        <input
          class="t-stepper__input"
          type="tel"
        />
        <div
          class="t-stepper__plus"
        >
          <svg
            class="t-icon t-icon-add t-stepper__icon"
            fill="none"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M7.35 8.65v3.85h1.3V8.65h3.85v-1.3H8.65V3.5h-1.3v3.85H3.5v1.3h3.85z"
              fill="currentColor"
              fill-opacity="0.9"
            />
          </svg>
        </div>
      </div>
    </div>
    
  </div>
</div>
`;
